base/src/org/adempiere/process/YearEndClosing.java

```diff
@@ -0,0 +1,183 @@
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.compiere.model.MAccount;
+import org.compiere.model.MAcctSchema;
+import org.compiere.model.MConversionType;
+import org.compiere.model.MElementValue;
+import org.compiere.model.MJournal;
+import org.compiere.model.MJournalBatch;
+import org.compiere.model.MJournalLine;
+import org.compiere.model.MPeriod;
+import org.compiere.model.Query;
+import org.compiere.util.DB;
+import org.compiere.util.Env;
+import org.compiere.util.Msg;
+ 
+
+/**
+ *	Balances expense/income account to a selected account
+ *	
+ *  @author Susanne Calderon
+ *  @version $Id: yearendclosing.java,v 1.2 2011/01/20$
+ */
+public class YearEndClosing extends YearEndClosingAbstract
+{
+	private int					closingAccountId = 0;
+	
+	/**
+	 *  Prepare - e.g., get Parameters.
+	 */
+	protected void prepare()
+	{
+		super.prepare();
+	}	//	prepare
+
+	/**
+	 *  Perform process.
+	 *  @return Message (clear text)
+	 *  @throws Exception if not successful
+	 */
+	protected String doIt () throws Exception
+	{
+		MAcctSchema acctSchema = new MAcctSchema(getCtx(), getAcctSchemaId(), get_TrxName());
+		closingAccountId = createAccount(getElementValueId());
+		if (closingAccountId == 0)
+			return "";
+		MJournalBatch journalBatch = new MJournalBatch(getCtx(), 0, get_TrxName());	
+		
+			
+		BigDecimal balance = Env.ZERO;
+		String documentno = Msg.translate(Env.getCtx(), "Closing")+ "_"  + getDateAcct().toString();
+		String description =  Msg.translate(Env.getCtx(), "Closing") + " "  + getDateAcct().toString();
+		journalBatch.setAD_Org_ID(getOrgId());
+		journalBatch.setDateAcct(getDateAcct());
+		journalBatch.setDateDoc(getDateAcct());
+		journalBatch.setDescription(documentno);
+		journalBatch.setC_DocType_ID(getDocTypeId());
+		journalBatch.setDocumentNo(description);
+		journalBatch.setGL_Category_ID(getCategoryId());
+		journalBatch.setControlAmt(Env.ZERO);
+		journalBatch.setC_Currency_ID(acctSchema.getC_Currency_ID());
+		journalBatch.save();
+		
+		String whereclause = "accounttype in ('E', 'R')  and ISSUMMARY = 'N' AND c_Element_ID=?";
+		int elementId = DB.getSQLValueEx(get_TrxName(), 
+				"SELECT C_Element_ID FROM C_AcctSchema_Element WHERE elementtype = 'AC' AND C_AcctSchema_ID=?", getAcctSchemaId());
+		List<MElementValue> accounts = new Query(getCtx(), MElementValue.Table_Name, 
+				whereclause, get_TrxName())
+		.setClient_ID()
+		.setParameters(elementId)
+		.setOrderBy("value")
+		.list();		
+		for (MElementValue elementValue : accounts) 
+		{			
+
+			log.info (elementValue.getValue());		
+			balance = searchBalance(elementValue.getC_ElementValue_ID());
+			if (balance.equals(Env.ZERO))
+				continue;
+			createJournal(elementValue, journalBatch, acctSchema, balance);
+		}
+		return "@Created@: " + journalBatch.getDocumentNo();		
+	}	//	doIt
+	
+	private int createAccount(int elementValueId)
+	{   
+		
+		
+		MAccount acc = MAccount.get(getCtx(), getAD_Client_ID(),

```

Comment:
@SusanneCalderon Can you name the variables with a complete word, usually, when I use Intellij the IDE suggests the best name based on an English dictionary, in this case, you can use account or account valid combination?